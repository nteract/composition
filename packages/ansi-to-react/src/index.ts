import * as React from "react";
import { ansiToJson, AnserJsonEntry } from "anser";
import { escapeCarriageReturn } from "escape-carriage";

const LINK_REGEX = /(https?:\/\/(?:www\.|(?!www))[^\s\.]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})/;

/**
 * Converts ANSI strings into JSON output.
 * @name ansiToJSON
 * @function
 * @param {String} input The input string.
 * @return {Array} The parsed input.
 */
function ansiToJSON(input: string) {
  input = escapeCarriageReturn(input);
  return ansiToJson(input, {
    json: true,
    remove_empty: true
  });
}

/**
 * Converts an Anser bundle into a React Node.
 * @param linkify whether links should be converting into clickable anchor tags.
 * @param bundle Anser output.
 */
function convertBundleIntoReact(
  linkify: boolean,
  bundle: AnserJsonEntry,
  key: number
) {
  const style: { backgroundColor?: string; color?: string } = {};
  if (bundle.bg) {
    style.backgroundColor = `rgb(${bundle.bg})`;
  }
  if (bundle.fg) {
    style.color = `rgb(${bundle.fg})`;
  }

  if (!linkify) {
    return React.createElement("span", { style, key }, bundle.content);
  }

  const words = bundle.content.split(" ").reduce(
    (words: React.ReactNode[], word: string, index: number) => {
      // If this isn't the first word, re-add the space removed from split.
      if (index !== 0) {
        words.push(" ");
      }

      // If  this isn't a link, just return the word as-is.
      if (!LINK_REGEX.test(word)) {
        words.push(word);
        return words;
      }

      words.push(
        React.createElement(
          "a",
          {
            key: index,
            href: word,
            target: "_blank"
          },
          `${word}`
        )
      );
      return words;
    },
    [] as React.ReactNode[]
  );
  return React.createElement("span", { style, key }, words);
}

declare type Props = {
  children: string;
  className?: string;
  linkify?: boolean;
};

export default function Ansi(props: Props) {
  return React.createElement(
    "code",
    { className: props.className },
    ansiToJSON(props.children).map(
      convertBundleIntoReact.bind(null, props.linkify)
    )
  );
}
