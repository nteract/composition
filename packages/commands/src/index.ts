import { COMMANDS } from "./commands";
import { makeCommandSetsComponent } from "./components";
import { CellTarget, CommandLocation, GapTarget, makeCellInfo, makeCellModelFromState, makeGapInfo, makeGapModel } from "./model";

export { COMMANDS };
export { CellTarget, CommandLocation, GapTarget, makeCellInfo, makeCellModelFromState, makeGapInfo, makeGapModel };

export const CellCommands =
  makeCommandSetsComponent(makeCellInfo, makeCellModelFromState, COMMANDS.CELL);

export const GapCommands =
  makeCommandSetsComponent(makeGapInfo, makeGapModel, COMMANDS.GAP);
