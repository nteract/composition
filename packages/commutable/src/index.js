/* @flow */

import type { Notebook as v4Notebook } from './v4';

import type {
  ImmutableNotebook,
  JSONType,
} from './types';

type PlaceholderNotebook = {
  nbformat: number,
  nbformat_minor: number,
};

const v4 = require('./v4');

const {
  emptyNotebook,
  emptyCodeCell,
  emptyMarkdownCell,
  appendCell,
  monocellNotebook,
} = require('./structures');

export type Notebook = PlaceholderNotebook & v4Notebook;

function freezeReviver(k: string, v: JSONType): JSONType {
  return Object.freeze(v);
}

// Expected usage of below is
// fromJS(parseNotebook(string|buffer))

function parseNotebook(notebookString: string): Notebook {
  return JSON.parse(notebookString, freezeReviver);
}

function fromJS(notebookJSON: Notebook): ImmutableNotebook {
  if (notebookJSON.nbformat === 4 && notebookJSON.nbformat_minor >= 0) {
    if (Array.isArray(notebookJSON.cells) && typeof notebookJSON.metadata === 'object') {
      return v4.fromJS(notebookJSON);
    }
  }

  if (notebookJSON.nbformat) {
    throw new TypeError(
      `nbformat v${notebookJSON.nbformat}.${notebookJSON.nbformat_minor} not recognized`
    );
  }

  throw new TypeError('This notebook format is not supported');
}

function toJS(immnb: ImmutableNotebook): v4Notebook {
  if (immnb.get('nbformat') === 4 && immnb.get('nbformat_minor') >= 0) {
    return v4.toJS(immnb);
  }
  throw new TypeError('Only notebook format 4 is supported');
}

// Expected usage is stringifyNotebook(toJS(immutableNotebook))
function stringifyNotebook(notebook: v4Notebook): string {
  return JSON.stringify(notebook, null, 2);
}

module.exports = {
  emptyCodeCell,
  emptyMarkdownCell,
  emptyNotebook,
  appendCell,
  monocellNotebook,
  toJS,
  fromJS,
  stringifyNotebook,
};

export type {
  ImmutableNotebook,
};
