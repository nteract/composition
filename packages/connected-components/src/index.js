// @flow

export {
  default as NewNotebookNavigation,
  PureNewNotebookNavigation
} from "./new-notebook-navigation";

export { default as NotebookMenu } from "./notebook-menu";
export { default as ModalController } from "./modal-controller";
