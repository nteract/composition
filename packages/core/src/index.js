//@flow

import * as actions from "./actions";
import * as constants from "./constants";
import * as middlewares from "./middlewares";
import * as records from "./records";
import * as reducers from "./reducers";
import * as components from "./components";

export { actions, constants, middlewares, records, reducers, components };
