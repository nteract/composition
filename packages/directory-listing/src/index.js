// @flow
import { Entry } from "./components/entry";
import { Listing } from "./components/listing";
import { Icon } from "./components/icon";
import { Name } from "./components/name";
import { LastSaved } from "./components/lastsaved";

export { Entry, Listing, Icon, Name, LastSaved };
