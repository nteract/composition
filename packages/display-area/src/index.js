// @flow
import Display from "./display";
import RichestMime from "./richest-mime";

module.exports = {
  Display,
  RichestMime
};
