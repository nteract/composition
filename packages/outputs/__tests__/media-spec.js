import * as React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import renderer from "react-test-renderer";

import { Media } from "../src";

describe("HTML", () => {
  it("renders direct HTML", () => {
    const component = mount(<Media.HTML data={"<b>cats are best</b>"} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it("updates the underlying HTML when data changes", () => {
    const component = mount(<Media.HTML data={"<b>cats are best</b>"} />);
    component.setProps({ data: "<b>squirrels are pretty great, too</b>" });
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe("<JSON />;", () => {
  const population = {
    total_population: [
      { date: "2018-10-13", population: 329499086 },
      { date: "2018-10-14", population: 329505562 }
    ]
  };

  it("renders JSON data", () => {
    const component = renderer
      .create(<Media.JSON data={population} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("updates the theme if it changes", () => {
    const component = renderer
      .create(<Media.JSON data={population} theme={"dark"} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe("<SVG />", () => {
  it("renders SVG data", () => {
    const data = `<svg
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
        width="96"
        height="20"
      >
        <linearGradient id="b" x2="0" y2="100%">
          <stop offset="0" stop-color="#bbb" stop-opacity=".1" />
          <stop offset="1" stop-opacity=".1" />
        </linearGradient>
        <clipPath id="a">
          <rect width="96" height="20" rx="3" fill="#fff" />
        </clipPath>
        <g clip-path="url(#a)">
          <path fill="#555" d="M0 0h47v20H0z" />
          <path fill="#8518f2" d="M47 0h49v20H47z" />
          <path fill="url(#b)" d="M0 0h96v20H0z" />
        </g>
        <g
          fill="#fff"
          text-anchor="middle"
          font-family="DejaVu Sans,Verdana,Geneva,sans-serif"
          font-size="110"
        >
          <text
            x="245"
            y="150"
            fill="#010101"
            fill-opacity=".3"
            transform="scale(.1)"
            textLength="370"
          >
            launch
          </text>
          <text x="245" y="140" transform="scale(.1)" textLength="370">
            launch
          </text>
          <text
            x="705"
            y="150"
            fill="#010101"
            fill-opacity=".3"
            transform="scale(.1)"
            textLength="390"
          >
            nteract
          </text>
          <text x="705" y="140" transform="scale(.1)" textLength="390">
            nteract
          </text>
        </g>
      </svg>`;
    const component = mount(<Media.SVG data={data} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe("Markdown", () => {
  it("Should render markdown", () => {
    const data = "# Header";
    const component = mount(<Media.Markdown data={data} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe("Plain", () => {
  it("Should render markdown", () => {
    const data = "The text in Spain is mainly plain";
    const component = mount(<Media.Plain data={data} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
