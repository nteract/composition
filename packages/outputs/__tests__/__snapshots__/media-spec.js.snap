// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Json /> renders JSON data 1`] = `
<ul
  style={
    Object {
      "MozUserSelect": "none",
      "WebkitUserSelect": "none",
      "backgroundColor": "transparent",
      "border": 0,
      "listStyle": "none",
      "marginBottom": "0.5em",
      "marginLeft": "0.125em",
      "marginRight": 0,
      "marginTop": "0.5em",
      "padding": 0,
    }
  }
>
  <li
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <ul
      style={
        Object {
          "listStyle": "none",
          "margin": 0,
          "padding": 0,
        }
      }
    >
      <li
        style={
          Object {
            "marginLeft": 0,
            "paddingLeft": 0,
            "paddingTop": "0.25em",
            "position": "relative",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "inline-block",
              "paddingLeft": 0,
              "paddingRight": "0.5em",
            }
          }
        >
          <div
            style={
              Object {
                "MozTransform": "rotateZ(90deg)",
                "MozTransformOrigin": "45% 50%",
                "MozTransition": "150ms",
                "WebkitTransform": "rotateZ(90deg)",
                "WebkitTransformOrigin": "45% 50%",
                "WebkitTransition": "150ms",
                "color": "#268bd2",
                "fontSize": "0.75em",
                "lineHeight": "1.1em",
                "marginLeft": 0,
                "position": "relative",
                "transform": "rotateZ(90deg)",
                "transformOrigin": "45% 50%",
                "transition": "150ms",
              }
            }
          >
            ▶
          </div>
        </div>
        <label
          onClick={[Function]}
          style={
            Object {
              "MozUserSelect": "inherit",
              "WebkitUserSelect": "inherit",
              "color": "#268bd2",
              "cursor": "pointer",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
            }
          }
        >
          <span>
            total_population
            :
          </span>
        </label>
        <span
          onClick={[Function]}
          style={
            Object {
              "color": "#657b83",
              "cursor": "default",
              "paddingLeft": "0.5em",
            }
          }
        >
          <span>
            <span
              style={
                Object {
                  "marginLeft": "0.3em",
                  "marginRight": "0.3em",
                }
              }
            >
              []
            </span>
             
            2 items
          </span>
        </span>
        <ul
          style={
            Object {
              "display": "block",
              "listStyle": "none",
              "margin": 0,
              "padding": 0,
            }
          }
        >
          <li
            style={
              Object {
                "marginLeft": "0.875em",
                "paddingLeft": 0,
                "paddingTop": "0.25em",
                "position": "relative",
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "paddingLeft": 0,
                  "paddingRight": "0.5em",
                }
              }
            >
              <div
                style={
                  Object {
                    "MozTransform": "rotateZ(90deg)",
                    "MozTransformOrigin": "45% 50%",
                    "MozTransition": "150ms",
                    "WebkitTransform": "rotateZ(90deg)",
                    "WebkitTransformOrigin": "45% 50%",
                    "WebkitTransition": "150ms",
                    "color": "#268bd2",
                    "fontSize": "0.75em",
                    "lineHeight": "1.1em",
                    "marginLeft": 0,
                    "position": "relative",
                    "transform": "rotateZ(90deg)",
                    "transformOrigin": "45% 50%",
                    "transition": "150ms",
                  }
                }
              >
                ▶
              </div>
            </div>
            <label
              onClick={[Function]}
              style={
                Object {
                  "MozUserSelect": "inherit",
                  "WebkitUserSelect": "inherit",
                  "color": "#268bd2",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <span>
                0
                :
              </span>
            </label>
            <span
              onClick={[Function]}
              style={
                Object {
                  "color": "#657b83",
                  "cursor": "default",
                  "paddingLeft": "0.5em",
                }
              }
            >
              <span>
                <span
                  style={
                    Object {
                      "marginLeft": "0.3em",
                      "marginRight": "0.3em",
                    }
                  }
                >
                  {}
                </span>
                 
                2 keys
              </span>
            </span>
            <ul
              style={
                Object {
                  "display": "block",
                  "listStyle": "none",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    date
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#66BB6A",
                    }
                  }
                >
                  "2018-10-13"
                </span>
              </li>
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    population
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#cb4b16",
                    }
                  }
                >
                  329499086
                </span>
              </li>
            </ul>
          </li>
          <li
            style={
              Object {
                "marginLeft": "0.875em",
                "paddingLeft": 0,
                "paddingTop": "0.25em",
                "position": "relative",
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "paddingLeft": 0,
                  "paddingRight": "0.5em",
                }
              }
            >
              <div
                style={
                  Object {
                    "MozTransform": "rotateZ(90deg)",
                    "MozTransformOrigin": "45% 50%",
                    "MozTransition": "150ms",
                    "WebkitTransform": "rotateZ(90deg)",
                    "WebkitTransformOrigin": "45% 50%",
                    "WebkitTransition": "150ms",
                    "color": "#268bd2",
                    "fontSize": "0.75em",
                    "lineHeight": "1.1em",
                    "marginLeft": 0,
                    "position": "relative",
                    "transform": "rotateZ(90deg)",
                    "transformOrigin": "45% 50%",
                    "transition": "150ms",
                  }
                }
              >
                ▶
              </div>
            </div>
            <label
              onClick={[Function]}
              style={
                Object {
                  "MozUserSelect": "inherit",
                  "WebkitUserSelect": "inherit",
                  "color": "#268bd2",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <span>
                1
                :
              </span>
            </label>
            <span
              onClick={[Function]}
              style={
                Object {
                  "color": "#657b83",
                  "cursor": "default",
                  "paddingLeft": "0.5em",
                }
              }
            >
              <span>
                <span
                  style={
                    Object {
                      "marginLeft": "0.3em",
                      "marginRight": "0.3em",
                    }
                  }
                >
                  {}
                </span>
                 
                2 keys
              </span>
            </span>
            <ul
              style={
                Object {
                  "display": "block",
                  "listStyle": "none",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    date
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#66BB6A",
                    }
                  }
                >
                  "2018-10-14"
                </span>
              </li>
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    population
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#cb4b16",
                    }
                  }
                >
                  329505562
                </span>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`<Json /> updates the theme if it changes 1`] = `
<ul
  style={
    Object {
      "MozUserSelect": "none",
      "WebkitUserSelect": "none",
      "backgroundColor": "transparent",
      "border": 0,
      "listStyle": "none",
      "marginBottom": "0.5em",
      "marginLeft": "0.125em",
      "marginRight": 0,
      "marginTop": "0.5em",
      "padding": 0,
    }
  }
>
  <li
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <ul
      style={
        Object {
          "listStyle": "none",
          "margin": 0,
          "padding": 0,
        }
      }
    >
      <li
        style={
          Object {
            "marginLeft": 0,
            "paddingLeft": 0,
            "paddingTop": "0.25em",
            "position": "relative",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "display": "inline-block",
              "paddingLeft": 0,
              "paddingRight": "0.5em",
            }
          }
        >
          <div
            style={
              Object {
                "MozTransform": "rotateZ(90deg)",
                "MozTransformOrigin": "45% 50%",
                "MozTransition": "150ms",
                "WebkitTransform": "rotateZ(90deg)",
                "WebkitTransformOrigin": "45% 50%",
                "WebkitTransition": "150ms",
                "color": "#268bd2",
                "fontSize": "0.75em",
                "lineHeight": "1.1em",
                "marginLeft": 0,
                "position": "relative",
                "transform": "rotateZ(90deg)",
                "transformOrigin": "45% 50%",
                "transition": "150ms",
              }
            }
          >
            ▶
          </div>
        </div>
        <label
          onClick={[Function]}
          style={
            Object {
              "MozUserSelect": "inherit",
              "WebkitUserSelect": "inherit",
              "color": "#268bd2",
              "cursor": "pointer",
              "display": "inline-block",
              "margin": 0,
              "padding": 0,
            }
          }
        >
          <span>
            total_population
            :
          </span>
        </label>
        <span
          onClick={[Function]}
          style={
            Object {
              "color": "#657b83",
              "cursor": "default",
              "paddingLeft": "0.5em",
            }
          }
        >
          <span>
            <span
              style={
                Object {
                  "marginLeft": "0.3em",
                  "marginRight": "0.3em",
                }
              }
            >
              []
            </span>
             
            2 items
          </span>
        </span>
        <ul
          style={
            Object {
              "display": "block",
              "listStyle": "none",
              "margin": 0,
              "padding": 0,
            }
          }
        >
          <li
            style={
              Object {
                "marginLeft": "0.875em",
                "paddingLeft": 0,
                "paddingTop": "0.25em",
                "position": "relative",
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "paddingLeft": 0,
                  "paddingRight": "0.5em",
                }
              }
            >
              <div
                style={
                  Object {
                    "MozTransform": "rotateZ(90deg)",
                    "MozTransformOrigin": "45% 50%",
                    "MozTransition": "150ms",
                    "WebkitTransform": "rotateZ(90deg)",
                    "WebkitTransformOrigin": "45% 50%",
                    "WebkitTransition": "150ms",
                    "color": "#268bd2",
                    "fontSize": "0.75em",
                    "lineHeight": "1.1em",
                    "marginLeft": 0,
                    "position": "relative",
                    "transform": "rotateZ(90deg)",
                    "transformOrigin": "45% 50%",
                    "transition": "150ms",
                  }
                }
              >
                ▶
              </div>
            </div>
            <label
              onClick={[Function]}
              style={
                Object {
                  "MozUserSelect": "inherit",
                  "WebkitUserSelect": "inherit",
                  "color": "#268bd2",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <span>
                0
                :
              </span>
            </label>
            <span
              onClick={[Function]}
              style={
                Object {
                  "color": "#657b83",
                  "cursor": "default",
                  "paddingLeft": "0.5em",
                }
              }
            >
              <span>
                <span
                  style={
                    Object {
                      "marginLeft": "0.3em",
                      "marginRight": "0.3em",
                    }
                  }
                >
                  {}
                </span>
                 
                2 keys
              </span>
            </span>
            <ul
              style={
                Object {
                  "display": "block",
                  "listStyle": "none",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    date
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#EDF3F7",
                    }
                  }
                >
                  "2018-10-13"
                </span>
              </li>
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    population
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#cb4b16",
                    }
                  }
                >
                  329499086
                </span>
              </li>
            </ul>
          </li>
          <li
            style={
              Object {
                "marginLeft": "0.875em",
                "paddingLeft": 0,
                "paddingTop": "0.25em",
                "position": "relative",
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "display": "inline-block",
                  "paddingLeft": 0,
                  "paddingRight": "0.5em",
                }
              }
            >
              <div
                style={
                  Object {
                    "MozTransform": "rotateZ(90deg)",
                    "MozTransformOrigin": "45% 50%",
                    "MozTransition": "150ms",
                    "WebkitTransform": "rotateZ(90deg)",
                    "WebkitTransformOrigin": "45% 50%",
                    "WebkitTransition": "150ms",
                    "color": "#268bd2",
                    "fontSize": "0.75em",
                    "lineHeight": "1.1em",
                    "marginLeft": 0,
                    "position": "relative",
                    "transform": "rotateZ(90deg)",
                    "transformOrigin": "45% 50%",
                    "transition": "150ms",
                  }
                }
              >
                ▶
              </div>
            </div>
            <label
              onClick={[Function]}
              style={
                Object {
                  "MozUserSelect": "inherit",
                  "WebkitUserSelect": "inherit",
                  "color": "#268bd2",
                  "cursor": "pointer",
                  "display": "inline-block",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <span>
                1
                :
              </span>
            </label>
            <span
              onClick={[Function]}
              style={
                Object {
                  "color": "#657b83",
                  "cursor": "default",
                  "paddingLeft": "0.5em",
                }
              }
            >
              <span>
                <span
                  style={
                    Object {
                      "marginLeft": "0.3em",
                      "marginRight": "0.3em",
                    }
                  }
                >
                  {}
                </span>
                 
                2 keys
              </span>
            </span>
            <ul
              style={
                Object {
                  "display": "block",
                  "listStyle": "none",
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    date
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#EDF3F7",
                    }
                  }
                >
                  "2018-10-14"
                </span>
              </li>
              <li
                style={
                  Object {
                    "MozUserSelect": "text",
                    "WebkitUserSelect": "text",
                    "marginLeft": "0.875em",
                    "paddingLeft": "2.125em",
                    "paddingRight": 0,
                    "paddingTop": "0.25em",
                    "textIndent": "-0.5em",
                    "wordBreak": "break-all",
                    "wordWrap": "break-word",
                  }
                }
              >
                <label
                  style={
                    Object {
                      "color": "#268bd2",
                      "display": "inline-block",
                      "margin": "0 0.5em 0 0",
                    }
                  }
                >
                  <span>
                    population
                    :
                  </span>
                </label>
                <span
                  style={
                    Object {
                      "color": "#cb4b16",
                    }
                  }
                >
                  329505562
                </span>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`<SVG /> renders SVG data 1`] = `
<SVG
  data="<svg
        xmlns=\\"http://www.w3.org/2000/svg\\"
        xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"
        width=\\"96\\"
        height=\\"20\\"
      >
        <linearGradient id=\\"b\\" x2=\\"0\\" y2=\\"100%\\">
          <stop offset=\\"0\\" stop-color=\\"#bbb\\" stop-opacity=\\".1\\" />
          <stop offset=\\"1\\" stop-opacity=\\".1\\" />
        </linearGradient>
        <clipPath id=\\"a\\">
          <rect width=\\"96\\" height=\\"20\\" rx=\\"3\\" fill=\\"#fff\\" />
        </clipPath>
        <g clip-path=\\"url(#a)\\">
          <path fill=\\"#555\\" d=\\"M0 0h47v20H0z\\" />
          <path fill=\\"#8518f2\\" d=\\"M47 0h49v20H47z\\" />
          <path fill=\\"url(#b)\\" d=\\"M0 0h96v20H0z\\" />
        </g>
        <g
          fill=\\"#fff\\"
          text-anchor=\\"middle\\"
          font-family=\\"DejaVu Sans,Verdana,Geneva,sans-serif\\"
          font-size=\\"110\\"
        >
          <text
            x=\\"245\\"
            y=\\"150\\"
            fill=\\"#010101\\"
            fill-opacity=\\".3\\"
            transform=\\"scale(.1)\\"
            textLength=\\"370\\"
          >
            launch
          </text>
          <text x=\\"245\\" y=\\"140\\" transform=\\"scale(.1)\\" textLength=\\"370\\">
            launch
          </text>
          <text
            x=\\"705\\"
            y=\\"150\\"
            fill=\\"#010101\\"
            fill-opacity=\\".3\\"
            transform=\\"scale(.1)\\"
            textLength=\\"390\\"
          >
            nteract
          </text>
          <text x=\\"705\\" y=\\"140\\" transform=\\"scale(.1)\\" textLength=\\"390\\">
            nteract
          </text>
        </g>
      </svg>"
  mediaType="image/svg+xml"
>
  <div />
</SVG>
`;

exports[`HTML renders direct HTML 1`] = `
<HTML
  data="<b>cats are best</b>"
  mediaType="text/html"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<b>cats are best</b>",
      }
    }
  />
</HTML>
`;

exports[`HTML updates the underlying HTML when data changes 1`] = `
<HTML
  data="<b>squirrels are pretty great, too</b>"
  mediaType="text/html"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<b>squirrels are pretty great, too</b>",
      }
    }
  />
</HTML>
`;

exports[`LaTeX Should render LaTeX 1`] = `
<LaTeX
  data="The well known Pythagorean theorem \\\\(x^2 + y^2 = z^2\\\\) was
    proved to be invalid for other exponents.
    Meaning the next equation has no integer solutions:

    \\\\[ x^n + y^n = z^n \\\\]"
  mediaType="text/latex"
>
  <MathJaxText
    onRender={null}
  >
    <Provider
      delay={0}
      didFinishTypeset={null}
      input="tex"
      loading={null}
      noGate={false}
      onError={[Function]}
      onLoad={null}
      options={Object {}}
      src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"
    >
      <MathJaxText
        onRender={null}
      />
    </Provider>
  </MathJaxText>
</LaTeX>
`;

exports[`Markdown Should render markdown 1`] = `
<Markdown
  data="# Header"
  mediaType="text/markdown"
>
  <MarkdownRender
    source="# Header"
  >
    <ReactMarkdown
      astPlugins={Array []}
      escapeHtml={false}
      plugins={
        Array [
          [Function],
        ]
      }
      rawSourcePos={false}
      renderers={
        Object {
          "inlineMath": [Function],
          "math": [Function],
        }
      }
      skipHtml={false}
      source="# Header"
      sourcePos={false}
      transformLinkUri={[Function]}
    >
      <Root
        key="root-1-1"
      >
        <Heading
          key="heading-1-1"
          level={1}
        >
          <h1>
            <TextRenderer
              key="text-1-3"
              nodeKey="text-1-3"
              value="Header"
            >
              Header
            </TextRenderer>
          </h1>
        </Heading>
      </Root>
    </ReactMarkdown>
  </MarkdownRender>
</Markdown>
`;

exports[`Plain Should render markdown 1`] = `
<Plain
  data="The text in Spain is mainly plain"
  mediaType="text/plain"
>
  <pre>
    <Ansi
      linkify={false}
    >
      <code>
        <span
          key="0"
          style={Object {}}
        >
          The text in Spain is mainly plain
        </span>
      </code>
    </Ansi>
  </pre>
</Plain>
`;
